src/volume_rendering/volume_render_layer.ts

```diff
@@ -1024,9 +1068,130 @@ outputValue = vec4(1.0, 1.0, 1.0, 1.0);
+              setControlsInShader(
+                gl,
+                shader,
+                this.shaderControlState,
+                shaderResult.parameters.parseResult.controls,
+              );
+              if (
+                renderContext.depthBufferTexture !== undefined &&
+                renderContext.depthBufferTexture !== null
+              ) {
+                const depthTextureUnit = shader.textureUnit(
+                  depthSamplerTextureUnit,
+                );
+                gl.activeTexture(
+                  WebGL2RenderingContext.TEXTURE0 + depthTextureUnit,
+                );
+                gl.bindTexture(
+                  WebGL2RenderingContext.TEXTURE_2D,
+                  renderContext.depthBufferTexture,
+                );
+              } else {
+                throw new Error(
+                  "Depth buffer texture ID for volume rendering is undefined or null",
+                );
+              }
+              chunkFormat.beginDrawing(gl, shader);
+              chunkFormat.beginSource(gl, shader);
+            }
+          }
+        }
+        if (shader === null) break;
+        if (chunkFormat != null) {
+          chunkFormat.bindChunk(
+            gl,
+            shader,
+            chunkInfo.chunk,
+            chunkInfo.fixedPositionWithinChunk,
+            chunkInfo.chunkDisplayDimensionIndices,
+            chunkInfo.channelToChunkDimensionIndices,
+            newSource,
+          );
+        }
+        gl.uniform3fv(shader.uniform("uTranslation"), uniforms.uTranslation);
+        gl.uniform3fv(
+          shader.uniform("uChunkDataSize"),
+          uniforms.uChunkDataSize,
+        );
+        // TODO (SKM) refactor to a new function
+        // Find the uniform with the same index
+        uniforms = shaderSetupUniformsForSecondPass;
+        gl.uniformMatrix4fv(
+          shader.uniform("uModelViewProjectionMatrix"),
+          false,
+          uniforms.uModelViewProjectionMatrix,
+        );
+        gl.uniformMatrix4fv(
+          shader.uniform("uInvModelViewProjectionMatrix"),
+          false,
+          uniforms.uInvModelViewProjectionMatrix,
+        );
+        gl.uniform1f(
+          shader.uniform("uNearLimitFraction"),
+          uniforms.uNearLimitFraction,
+        );
+        gl.uniform1f(
+          shader.uniform("uFarLimitFraction"),
+          uniforms.uFarLimitFraction,
+        );
+        gl.uniform1f(shader.uniform("uGain"), uniforms.uGain);
+        gl.uniform1ui(shader.uniform("uPickId"), uniforms.uPickId);
+        gl.uniform1i(shader.uniform("uMaxSteps"), uniforms.uMaxSteps);
+        gl.uniform3fv(
+          shader.uniform("uLowerClipBound"),
+          uniforms.uLowerClipBound,
+        );
+        gl.uniform3fv(
+          shader.uniform("uUpperClipBound"),
+          uniforms.uUpperClipBound,
+        );
+        drawBoxes(gl, 1, 1);
+        newSource = false;
+      }
+      this.modeOverride.value = VolumeRenderingModes.OFF;
```

Comment:
Is there a possible way the mode override can get stuck on here? Maybe a different way